src/config/ConfigOptions.ts

```diff
@@ -19,21 +19,24 @@ export interface ConfigOptions {
   rageshake?: {
     submit_url: string;
   };
-}
 
-export interface ResolvedConfigOptions extends ConfigOptions {
```

Comment:
I intended to use the `ConfigOptions` vs `ResolvedConfigOptions` distinction to keep ahead of strict mode errors. Because we always merge the default config with the downloaded config, the result of this merge had a more specific type than just `ConfigOptions`, which can help with null safety around the optional parts of the config - could we please not remove this?

Reply:
Ah, I thought it was probably something like this, although it was only shallow merging so it didn't seem like it was doing anything.

Reply:
Actually - doesn't it work if we just make the types non-nullable in ConfigOptions itself? I think the only reason we had to have them nullable before was because we returned an empty config object if there was no config, where we should have been returning the default.

Reply:
I personally think it still makes sense to keep the version of `ConfigOptions` with optional values, to type `downloadConfig` accurately which can return partial configs, and as documentation for what shape a config file is expected to have (ConfigOptions.ts is linked from the README), but none of that's critical.

Reply:
Fair enough - have added it back with a comment.